### PR TITLE
WSpinny: add makeCurrent() call in render()

### PR DESCRIPTION
**Pull request by Be-ing**
This improves performance on macOS.

### CODE REVIEW

**Comment by Be-ing** [comment]
@foss- can you test this with both OpenGL and non-OpenGL waveforms?

File: src/widget/wspinny.cpp [hunk 1]
```diff
@@ -318,6 +318,8 @@ void WSpinny::render(VSyncThread* vSyncThread) {
         return;
     }
 
+    makeCurrent();
```
Comment (Holzhaus) [inline src/widget/wspinny.cpp:321]: IIRC this is expensive. Everywhere else we use this:

```cpp
if (QGLContext::currentContext() != context()) {
    makeCurrent();
}
```

Comment (Be-ing) [reply src/widget/wspinny.cpp:321]: I tried that but it makes no difference. The condition is always false.

**Comment by foss-** [comment]
Tested macOS 10.15.7, 2017 MacBook Pro 3,1 GHz Quad-Core Intel Core i7, Radeon Pro 560 4 GB, Intel HD Graphics 630 1536 MB.

OpenGL-Status: 2.1 (AMD Radeon Pro 560 OpenGL Engine)
Sitting idle: ~44% CPU (fans already going)
Playing single track: ~99% CPU
Everything is very laggy. Opening preferences takes many seconds, dragging preferences window around is very laggy. Navigating preferences is slow. Clicking "Cancel" takes a while to be recognized. Only upside: audio is not loggy the track keeps playing fine. But any UI interaction is very painful.

Should there be an option do disable OpenGL?

**Comment by Holzhaus** [comment]
> Should there be an option do disable OpenGL?

There is, you can run Mixxx on the terminal with `--safeMode`. That will disable OpenGL. Does this "fix" the performance issues for you?

**Comment by foss-** [comment]
`/Applications/Mixxx.app/Contents/MacOS/mixxx --safeMode

no waveform shown
Sitting idle: ~5%
Playing single track ~17%
UI laggyness is gone

**Comment by Be-ing** [comment]
Okay. I give up trying to make spinnies perform well on macOS. The only path forward I see for 2.3 is to disable them by default.

**Pull request closed without merging**